rq/job.py

```diff
@@ -733,4 +732,38 @@ def register_dependency(self, pipeline=None):
             connection.sadd(dependents_key, self.id)
             connection.sadd(self.dependencies_key, dependency_id)
 
+    def dependencies_finished(
+            self,
+            pipeline=None
+    ):
+        """Returns a boolean indicating if all of this jobs dependencies are _FINISHED_
+
+        If a pipeline is passed, all dependencies are WATCHed.
+        """
+
+        pipe = pipeline if pipeline is not None else self.connection
+
+        if pipeline is not None:
+            pipe.watch(*[Job.key_for(as_text(_id))
+                         for _id in self.connection.smembers(self.dependencies_key)])
+
+        sort_by = self.redis_job_namespace_prefix + '*->ended_at'
+        get_field = self.redis_job_namespace_prefix + '*->status'
+
+        # As a minor optimization to more quickly tell if all dependencies
+        # are _FINISHED_, sort dependencies by the `ended_at` timestamp so
+        # those jobs which are not yet finished are at the start of the
+        # list. Sorting here lexographically works because these dates are
+        # stored in an ISO 8601 format, so lexographic order is the same as
+        # chronological order.
+        dependencies_statuses = [
+            (as_text(_id), as_text(status))
+            for _id, status in pipe.sort(name=self.dependencies_key, by=sort_by,
+                                         get=['#', get_field], alpha=True, groups=True, )
+        ]
+
+        return all(status == JobStatus.FINISHED
```

Comment:
We also need to account for situations where jobs are already finished but are no longer in Redis. In this case we can assume that the job is done.

Reply:
Just a reminder that this comment this needs to be addressed

Reply:
Addressed in d1ce1b26f8bc60b76f92d806add01096f3c29b31